resolwe/flow/executors/run.py

```diff
@@ -153,6 +153,24 @@ def update_data_status(self, **kwargs):
         })
 
     def run(self, data_id, script, verbosity=1):
+        """Execute the script and save results."""
+        try:
+            finish_fields = self._run(data_id, script, verbosity=verbosity)
+        except Exception as error:  # pylint: disable=broad-except
+            logger.exception("Unhandled exception in executor")
+
+            # Send error report.
+            self.update_data_status(process_error=[str(error)], status=DATA_META['STATUS_ERROR'])
+
+            finish_fields = {
+                ExecutorProtocol.FINISH_PROCESS_RC: 1,
+            }
+
+        if finish_fields is not None:
+            self._send_manager_command(ExecutorProtocol.FINISH, extra_fields=finish_fields)
+        # the feedback key deletes itself once the list is drained
```

Comment:
I found it confusing as well. It seems to be talking about the Redis list, which is deleted when all elements are popped? @jberci can you confirm this is the case?

Reply:
It is the case, the comment is about Redis deleting the list once all items have been popped. This behaviour was not obvious to me when I was doing this. If memory serves, the code was also structured so that one would expect explicit key deletion at the end, so I put that comment there. Feel free to get rid of it.

Reply:
Thanks for the explanation. I like @kostko's rewording and I think we can leave it in.